tests/snmp/test_snmp_loopback.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+import time
+import logging
+import ipaddress
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.device_type('vs')
+]
+
+@pytest.mark.bsl
+def test_snmp_loopback(duthosts, dut_index, nbrhosts, tbinfo, localhost, creds):
+    """
+    Test SNMP query to DUT over loopback IP 
+
+      - Send SNMP query over loopback IP from one of the BGP Neighbors
+      - Get SysDescr from snmpfacts 
+      - compare result from snmp query over loopback IP and snmpfacts
+
+    """
+    duthost = duthosts[dut_index]
+    hostip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    snmp_facts = localhost.snmp_facts(host=hostip, version="v2c", community=creds["snmp_rocommunity"])['ansible_facts']
+    config_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
+    # Get first neighbor VM information
+    nbr = nbrhosts[list(nbrhosts.keys())[0]]
+
+    # Perform SNMP walk rom neighbor  
+    for ip in config_facts[u'LOOPBACK_INTERFACE'][u'Loopback0']:
```

Comment:
>Loopback0 [](start = 52, length = 9)

Do we have test case for management port?
Both cases can share the code, and in future we may have either one plus ipv4 or ipv6. Reuse case will make future extension easier.

Reply:
The other SNMP test cases use management IP to query. The other SNMP test cases uses ansible snmp_facts module which will retrieve results over management IP. So in this test case, we specifically use IPv4 and IPv6 Loopback addresses present in config_db to query sysDescr OID. We can query any OID here, but chose one of the OIDs and compare it with information from snmp_facts.

Reply:
I see loopback ipv6 is covered here. But mgmt ipv6 is not covered by any test cases. Could you cover it in this PR?

---
In reply to: [518449100](https://github.com/Azure/sonic-mgmt/pull/2400#discussion_r518449100) [](ancestors = 518449100)

Reply:
@qiluo-msft I have updated the test case to use only IPv4 Loopback IP.
The query over Ipv6 Loopback Interface / Management Interface does not work.
I have raised a  github issue to look into this: https://github.com/Azure/sonic-snmpagent/issues/180